feature(oidc): implement oidc provider/op (#1811)
authelia/authelia: docs/configuration/identity-providers/oidc.md

---
layout: default
title: OpenID Connect
parent: Identity Providers
grand_parent: Configuration
nav_order: 2
---

# OpenID Connect

> **Authelia** currently supports [OpenID Connect] but is preview status. This means it's suggested you only implement

Reviewer: ```suggestion
**Authelia** currently supports [OpenID Connect] but is in beta version. This means it's suggested you only implement

Author: ```suggestion
**Authelia** currently supports [OpenID Connect] as a beta feature. This means it's suggested you only implement
```

Maybe this instead?

Reviewer: yep, fine.

it with caution. The main purpose of it being available is for us to allow users to try it and provide feedback. The 
reason we do it this way is [OpenID Connect] is a complicated technology to implement well, and we are more likely to
get good feedback if we allow people to test it. By default [OpenID Connect] is disabled unless you configure it.

```yaml
identity_providers:
  oidc:
    hmac_secret: this_is_a_secret_abc123abc123abc
    issuer_private_key: |
      --- KEY START
      --- KEY END
    clients:
      - id: myapp
        description: My Application
        secret: this_is_a_secret
        policy: two_factor
        redirect_uris:
          - https://oidc.example.com:8080/oauth2/callback
        scopes:
          - openid
          - groups
          - email
          - profile
        grant_types:
          - refresh_token
          - authorization_code
        response_types:
          - code
```

## Options

> ### hmac_secret

Reviewer: Needs to be updated if we go for `signing_key` or the like.
